Mathlib/RepresentationTheory/GroupCohomology/LowDegree.lean

```diff
@@ -332,19 +347,285 @@ theorem oneCoboundaries_eq_bot_of_isTrivial (A : Rep k G) [A.IsTrivial] :
+    IsTwoCoboundary f ↔ ∃ x : G → A, ∀ g h : G, g • x h - x (g * h) + x g = f (g, h) := Iff.rfl
+
+end IsCoboundary
+section ofDistribMulAction
+
+variable {k G A : Type u} [CommRing k] [Group G] [AddCommGroup A] [Module k A]
+  [DistribMulAction G A] [SMulCommClass G k A]
+
+/-- Given a `k`-module `A` with a compatible `DistribMulAction` of `G`, and a function
+`f : G → A` satisfying the 1-cocycle condition, produces a 1-cocycle for the representation on
+`A` induced by the `DistribMulAction`. -/
+def oneCocyclesOfIsOneCocycle (f : G → A) (hf : IsOneCocycle f) :
+    oneCocycles (Rep.ofDistribMulAction k G A) :=
+  ⟨f, (mem_oneCocycles_iff (A := Rep.ofDistribMulAction k G A) f).2 hf⟩
+
+theorem isOneCocycle_of_oneCocycles (f : oneCocycles (Rep.ofDistribMulAction k G A)) :
+    IsOneCocycle (A := A) f.1 := (mem_oneCocycles_iff f.1).1 f.2
+
+/-- Given a `k`-module `A` with a compatible `DistribMulAction` of `G`, and a function
+`f : G → A` satisfying the 1-coboundary condition, produces a 1-coboundary for the representation
+on `A` induced by the `DistribMulAction`. -/
+def oneCoboundariesOfIsOneCoboundary (f : G → A) (hf : IsOneCoboundary f) :
+    oneCoboundaries (Rep.ofDistribMulAction k G A) :=
+  oneCoboundariesOfMemRange f (by rcases hf with ⟨x, hx⟩; exact ⟨x, by ext g; exact hx g⟩)
+
+theorem isOneCoboundary_of_oneCoboundaries (f : oneCoboundaries (Rep.ofDistribMulAction k G A)) :
+    IsOneCoboundary (A := A) f.1.1 := by
+  rcases mem_range_of_mem_oneCoboundaries f.1 f.2 with ⟨x, hx⟩
+  exact ⟨x, by rw [← hx]; intro g; rfl⟩
+
+/-- Given a `k`-module `A` with a compatible `DistribMulAction` of `G`, and a function
+`f : G × G → A` satisfying the 2-cocycle condition, produces a 2-cocycle for the representation on
+`A` induced by the `DistribMulAction`. -/
+def twoCocyclesOfIsTwoCocycle (f : G × G → A) (hf : IsTwoCocycle f) :
+    twoCocycles (Rep.ofDistribMulAction k G A) :=
+  ⟨f, (mem_twoCocycles_iff (A := Rep.ofDistribMulAction k G A) f).2 hf⟩
+
+theorem isTwoCocycle_of_twoCocycles (f : twoCocycles (Rep.ofDistribMulAction k G A)) :
+    IsTwoCocycle (A := A) f.1 := (mem_twoCocycles_iff f.1).1 f.2
+
+/-- Given a `k`-module `A` with a compatible `DistribMulAction` of `G`, and a function
+`f : G × G → A` satisfying the 2-coboundary condition, produces a 2-coboundary for the
+representation on `A` induced by the `DistribMulAction`. -/
+def twoCoboundariesOfIsTwoCoboundary (f : G × G → A) (hf : IsTwoCoboundary f) :
+    twoCoboundaries (Rep.ofDistribMulAction k G A) :=
+  twoCoboundariesOfMemRange f (by rcases hf with ⟨x, hx⟩; exact ⟨x, by ext g; exact hx g.1 g.2⟩)
+
+theorem isTwoCoboundary_of_twoCoboundaries (f : twoCoboundaries (Rep.ofDistribMulAction k G A)) :
+    IsTwoCoboundary (A := A) f.1.1 := by
+  rcases mem_range_of_mem_twoCoboundaries f.1 f.2 with ⟨x, hx⟩
+  exact ⟨x, fun g h => Function.funext_iff.1 hx (g, h)⟩
+
+end ofDistribMulAction
+section IsMulCocycle
+section
+variable {G M : Type*} [Mul G] [CommGroup M] [SMul G M]
```

Comment:
```suggestion
end ofDistribMulAction

section IsMulCocycle

section

variable {G M : Type*} [Mul G] [CommGroup M] [SMul G M]
```
This section is the same as above but for multiplicative groups, right? If this is the case we probably want a docstring explaining it.